File: src/main/kotlin/org/gameflow/state/GameState.kt
package org.gameflow.state

import org.gameflow.Updating
import org.gameflow.Game
import org.gameflow.Rendering
import org.gameflow.entity.EntityGroups
import org.gameflow.pass.Pass
import org.gameflow.entity.Entity


/**
 * 
 */
public trait GameState: Entity {

    val entityGroups: EntityGroups

    val name: String

    fun onEnter(game: Game)
    fun onExit(game: Game)

}

File: src/main/scala/org/gameflow/core/state/GameState.scala
package org.gameflow.core.state

import org.gameflow.core.entity.{EntityGroupsImpl, EntityGroups}
import org.gameflow.core.pass.Passable
import org.gameflow.core.{LifeCycled, Game}


/**
 *
 */
trait GameState extends Passable with LifeCycled {

  def name: Symbol

  val groups: EntityGroups = new EntityGroupsImpl()

  def onEnter(game: Game) {}
  def onExit(game: Game) {}

}